lib/rules/no-unused-vars.js

```diff
@@ -372,22 +374,18 @@ module.exports = {
             return ref.isRead() && (
 
                 // self update. e.g. `a += 1`, `a++`
-                (
-                    parent.type === "AssignmentExpression" &&
+                (// in RHS of an assignment for itself. e.g. `a = a + 1`
+                    ((
```

Comment:
The indentation and extra parentheses are a little confusing here. (Was this change intentional? I see a comment was moved as well...)